src/traces/image/plot.js

```diff
@@ -13,14 +13,24 @@ var Lib = require('../../lib');
 var xmlnsNamespaces = require('../../constants/xmlns_namespaces');
 var constants = require('./constants');
 
+function compatibleAxis(ax) {
+    return ax.type === 'linear' &&
+        // y axis must be reversed but x axis mustn't be
+        ((ax.range[1] > ax.range[0]) === (ax._id.charAt(0) === 'x'));
+}
+
 module.exports = function plot(gd, plotinfo, cdimage, imageLayer) {
     var xa = plotinfo.xaxis;
     var ya = plotinfo.yaxis;
 
+    var supportsPixelatedImage = !Lib.isIOS() && !Lib.isSafari() && !Lib.isIE && !gd._context._exportedPlot;
+
     Lib.makeTraceGroups(imageLayer, cdimage, 'im').each(function(cd) {
         var plotGroup = d3.select(this);
         var cd0 = cd[0];
         var trace = cd0.trace;
+        var fastImage = supportsPixelatedImage && trace._isFromSource && compatibleAxis(xa) && compatibleAxis(ya);
+        trace._fastImage = fastImage;
```

Comment:
Hmm...
This `_fastImage` property appears to only be used in tests?!

Reply:
> This `_fastImage` property appears to only be used in tests?!

That's correct. It also helps with debugging.